botocore/parsers.py

```diff
@@ -707,8 +720,10 @@ def _do_error_parse(self, response, shape):
             # code has a couple forms as well:
             # * "com.aws.dynamodb.vAPI#ProvisionedThroughputExceededException"
             # * "ResourceNotFoundException"
+            if ':' in code:
+                code = code.split(':', 1)[0]
```

Comment:
There are now 3 distinct `if` statements here that aren't using `elif/else`. Is the intent to continually mutate the `code` value as it goes through each check?

e.g.

`com.test.mypackage#this:that#1` -- `split(':', 1)` --> `com.test.mypackage#this`
`com.test.mypackage#` -- `split('#', 1)` --> `this`

Then `this` is evaluated for the `x-amzn-query-error` code?